pkg/storage/etcd/etcd_watcher_test.go

```diff
@@ -451,7 +463,7 @@ func TestWatchFromZeroIndex(t *testing.T) {
 		t.Errorf("Unexpected event %#v", event)
 	}
 
-	if e, a := pod, event.Object; !api.Semantic.DeepDerivative(e, a) {
+	if e, a := pod, event.Object; a != nil && !api.Semantic.DeepDerivative(e, a) {
```

Comment:
why the nil check? if a is nil, this skips the equality check, which seems wrong